ICSharpCode.CodeConverter/CSharp/MethodBodyExecutableStatementVisitor.cs

```diff
@@ -587,7 +588,18 @@ public override async Task<SyntaxList<StatementSyntax>> VisitSelectBlock(VBSynta
                     csBlockStatements.Add(SyntaxFactory.BreakStatement());
                 }
                 var list = SingleStatement(SyntaxFactory.Block(csBlockStatements));
-                sections.Add(SyntaxFactory.SwitchSection(SyntaxFactory.List(labels), list));
+                var labelDesc = string.Join(" ", labels.Select(l => l.ToFullString()));
```

Comment:
ToFullString would include trivia, ToString would be better. Though technically there are several ways of writing equivalent cases, such as using hex, binary or octal representations of numbers. We only need to deal with constants for the sake of our purposes here (since others will become pattern match cases), so using _semanticModel.GetConstantValue might be best.